crates/polars-core/src/chunked_array/ops/compare_inner.rs

```diff
@@ -77,6 +78,18 @@ impl_traits!(BoolTakeRandomSingleChunk<'_>);
 impl_traits!(NumTakeRandomSingleChunk<'_, T>, T);
 impl_traits!(NumTakeRandomChunked<'_, T>, T);
 
+impl<'a> PartialEqInner for ListTakeRandomSingleChunk<'a> {
+    unsafe fn eq_element_unchecked(&self, idx_a: usize, idx_b: usize) -> bool {
+        self.get_unchecked(idx_a).map(Wrap) == self.get_unchecked(idx_b).map(Wrap)
```

Comment:
Compare for two `Series` may have a lot of overhead. Perhaps we can reimplement a `PartialEqInner` specifically designed for `List` as a key, and directly comparing based on `&[u8]` instead of `Series`. But I didn't carefully check if this way has any traps. So I didn't do that directly. Anyway, I'd like to ask for your opinion.

Reply:
Yes, this is definitively expensive. 

I think for this PR this is good. First get the operation to work without panicking. For a following PR, I think it might be interesting to use the `row_encoding` to convert all columns (if there are lists) to a single row-encode and then group the row-encoded `BinaryChunked` column.

Reply:
Yeah, that's make sense.  Next, let me think about how to make this better. :)